File: examples/sample.py
import time
import sys
from cryptlex.lexfloatclient import LexFloatClient, LexFloatStatusCodes, LexFloatClientException
# from cryptlex.lexfloatclient import *


def licence_callback_fn(status):
    if LexFloatStatusCodes.LF_OK == status:
        print("The license lease has renewed successfully.")
    elif LexFloatStatusCodes.LF_E_LICENSE_NOT_FOUND == status:
        print("The license expired before it could be renewed.")
    elif LexFloatStatusCodes.LF_E_LICENSE_EXPIRED_INET == status:
        print("The license expired due to network connection failure.")
    else:
        print("The license renew failed due to other reason. Error code: ", status)


def main():
    try:
        # Set the product id
        LexFloatClient.SetHostProductId("PASTE_PRODUCT_ID")
        # Set the float server
        LexFloatClient.SetHostUrl("http://localhost:8090")
        # Set the license callback
        LexFloatClient.SetFloatingLicenseCallback(licence_callback_fn)
        # Request license lease
        LexFloatClient.RequestFloatingLicense()
        print("Success! License acquired. Press enter to get the license metadata...")
        sys.stdin.read(1)
        # Request license metadata
        # metadata_value = LexFloatClient.GetHostLicenseMetadata("key1")
        print("Press enter to drop the license ...")
        sys.stdin.read(1)
        # Drop license lease
        LexFloatClient.DropFloatingLicense()
        print("Success! License dropped.")
    except LexFloatClientException as exception:
        print('Error code:', exception.code, exception.message)


main()
input("Press Enter to continue...")

File: examples/sample.go
package main

import (
	"bufio"
	"fmt"
	"github.com/cryptlex/lexfloatclient-go"
	"os"
)

func licenseCallback(status int) {
	if status == lexfloatclient.LF_OK {
		fmt.Println("The license lease has renewed successfully.")
	} else if status == lexfloatclient.LF_E_LICENSE_NOT_FOUND {
		fmt.Println("he license expired before it could be renewed.")
	} else if status == lexfloatclient.LF_E_LICENSE_EXPIRED_INET {
		fmt.Println("The license expired due to network connection failure.")
	} else {
		fmt.Println("The license renew failed due to other reason. Error code:", status)
	}
}

func main() {
	var status int
	status = lexfloatclient.SetHostProductId("PASTE_PRODUCT_ID")
	if lexfloatclient.LF_OK != status {
		fmt.Println("Error Code:", status)
		os.Exit(1)
	}
	status = lexfloatclient.SetHostUrl("http://localhost:8090")
	if lexfloatclient.LF_OK != status {
		fmt.Println("Error Code:", status)
		os.Exit(1)
	}
	lexfloatclient.SetFloatingLicenseCallback(licenseCallback)
	status = lexfloatclient.RequestFloatingLicense()
	if lexfloatclient.LF_OK != status {
		fmt.Println("Error Code:", status)
		os.Exit(1)
	}  
	fmt.Println("Success! License acquired.")
	var metadataValue string
    status = lexfloatclient.GetHostLicenseMetadata("key1", &metadataValue)
	if lexfloatclient.LF_OK != status {
		fmt.Println("Error getting license metadata. Code:", status)
	}  
	fmt.Println(metadataValue)
    status = lexfloatclient.DropFloatingLicense()
	if lexfloatclient.LF_OK != status {
		fmt.Println("Dropping license error code:", status)
	}  
    fmt.Println("Success! License dropped.")
	fmt.Println("Press any key to exit...")
	bufio.NewReader(os.Stdin).ReadByte()
}